tests/unit/test_node_latency_collector.py

```diff
@@ -0,0 +1,303 @@
+import random
+import time
+from concurrent.futures import ThreadPoolExecutor, wait
+from unittest.mock import patch
+
+import pytest
+from eth_typing import ChecksumAddress
+
+from nucypher.utilities.latency import NodeLatencyStatsCollector
+
+
+@pytest.fixture(scope="module")
+def execution_data(get_random_checksum_address):
+    executions = {}
+
+    node_1 = get_random_checksum_address()
+    node_1_exec_times = [11.23, 24.8, 31.5, 40.21]
+    executions[node_1] = node_1_exec_times
+
+    node_2 = get_random_checksum_address()
+    node_2_exec_times = [5.03, 6.78, 7.42, 8.043]
+    executions[node_2] = node_2_exec_times
+
+    node_3 = get_random_checksum_address()
+    node_3_exec_times = [0.44, 4.512, 3.3]
+    executions[node_3] = node_3_exec_times
+
+    node_4 = get_random_checksum_address()
+    node_4_exec_times = [0.1, 0.2, 0.3, 0.4, 0.5]
+    executions[node_4] = node_4_exec_times
+
+    sorted_order = sorted(
+        list(executions.keys()),
+        key=lambda x: sum(executions.get(x)) / len(executions.get(x)),
+    )
+    assert sorted_order == [
+        node_4,
+        node_3,
+        node_2,
+        node_1,
+    ]  # test of the test - "that's sooo meta"
+
+    return executions, sorted_order
+
+
+def floats_sufficiently_equal(a: float, b: float):
+    if abs(a - b) < 1e-9:
+        return True
+
+    return False
```

Comment:
```suggestion
    return abs(a - b) < 1e-9
```
But perhaps a better approach is to use `math.isclose(a, b)`:
https://docs.python.org/3/library/math.html#math.isclose

Reply:
✔️  - nice find!